### PR TITLE
fix dump data columns sort

### PR DESCRIPTION
**Pull request by fifsky**
The currently dumped data columns are out of order, which can be guaranteed by using yaml.MapSlice and yaml.MapItem

### CODE REVIEW

**Comment by fifsky** [comment]
And It also fixes the incorrect export data column types, especially integer types

File: dump.go [hunk 1]
```diff
@@ -97,7 +97,14 @@ func (d *Dumper) Dump() error {
...
 	query := fmt.Sprintf("SELECT * FROM %s", d.helper.quoteKeyword(table))
-	rows, err := d.db.Query(query)
+
+	stmt, err := d.db.Prepare(query)
+	if err != nil {
+		return err
+	}
+	defer stmt.Close()
+
+	rows, err := stmt.Query()
```
Comment (andreynering) [inline dump.go:107]: @fifsky I don't get why this change is necessary. Can you explain me?

**Review by andreynering — COMMENTED** [review]
Hi @fifsky, and thanks for opening this PR!

> And It also fixes the incorrect export data column types, especially integer types

Sorry, but I didn't get what you mean here.

File: dump.go (hunk 1 above)
Comment (fifsky) [reply dump.go:107]: The currently exported yml exports the integer as a string type, such as a primary key ID
```yml
- id: "1"    # primary key 
  user_id: "1"  
  content: test
  created_at: 2019-03-06T15:05:46+08:00
  updated_at: 2019-03-06T15:05:46+08:00
```
To resolve this issue, Must use mysql Prepare,This will give you the correct type
```yml
- id: 1    # primary key 
  user_id: 1 
  content: test
  created_at: 2019-03-06T15:05:46+08:00
  updated_at: 2019-03-06T15:05:46+08:00
```

Comment (andreynering) [reply dump.go:107]: I think that shouldn't happen anyway, but no problem

**Comment by andreynering** [comment]
This was just released as v3.1.2
